static/app/components/forms/fields/projectMapperField.tsx

```diff
@@ -173,80 +194,25 @@ export class RenderField extends Component<RenderProps, State> {
-      // if no value set, we want to return the default component that is rendered
-      const project = sentryProjectsById[selectedSentryProjectId || ''];
-      if (!project) {
-        return <components.ValueContainer {...containerProps} />;
-      }
-      return (
-        <components.ValueContainer {...containerProps}>
-          <IdBadge
-            project={project}
-            avatarSize={20}
-            displayName={project.slug}
-            avatarProps={{consistentWidth: true}}
-            disableLink
-          />
-        </components.ValueContainer>
-      );
-    };
-
-    const customOptionProject = projectProps => {
-      const project = sentryProjectsById[projectProps.value];
-      // Should never happen for a dropdown item
-      if (!project) {
-        return null;
-      }
-      return (
-        <components.Option {...projectProps}>
-          <IdBadge
-            project={project}
-            avatarSize={20}
-            displayName={project.slug}
-            avatarProps={{consistentWidth: true}}
-            disableLink
-          />
-        </components.Option>
-      );
-    };
-
-    const customMappedValueContainer = containerProps => {
-      // if no value set, we want to return the default component that is rendered
-      const mappedValue = mappedItemsByValue[selectedMappedValue || ''];
-      if (!mappedValue) {
-        return <components.ValueContainer {...containerProps} />;
-      }
-      return (
-        <components.ValueContainer {...containerProps}>
-          <IntegrationIconWrapper>{getIcon(iconType)}</IntegrationIconWrapper>
-          <OptionLabelWrapper>{mappedValue.label}</OptionLabelWrapper>
-        </components.ValueContainer>
-      );
-    };
-
-    const customOptionMappedValue = optionProps => {
-      return (
-        <components.Option {...optionProps}>
-          <OptionWrapper>
-            <IntegrationIconWrapper>{getIcon(iconType)}</IntegrationIconWrapper>
-            <OptionLabelWrapper>{optionProps.label}</OptionLabelWrapper>
-          </OptionWrapper>
-        </components.Option>
-      );
-    };
-
     return (
       <Fragment>
         {existingValues.map(renderItem)}
         <Item>
           <SelectControl
             placeholder={mappedValuePlaceholder}
             name="mappedDropdown"
-            options={mappedItemsToShow}
+            options={mappedItemOptions}
             components={{
-              Option: customOptionMappedValue,
-              ValueContainer: customMappedValueContainer,
+              SingleValue: containerProps => {
+                return (
+                  <components.ValueContainer {...containerProps}>
+                    <MappedValueContainer>
+                      {getIcon(iconType)}
+                      {mappedItemsByValue[selectedMappedValue || ''].label}
+                    </MappedValueContainer>
+                  </components.ValueContainer>
+                );
```

Comment:
i don't think we need these anymore?

Reply:
oh it kinda lines stuff up. hmmm

Reply:
what if we did something more like this and got the props out of the containerProps
```suggestion
              SingleValue: containerProps => {
                return (
                  <components.ValueContainer {...containerProps}>
                    <MappedValueContainer>
                      {containerProps.data.leadingItems}
                      {containerProps.children}
                    </MappedValueContainer>
                  </components.ValueContainer>
                );
```